src/library/Cardano/CoinSelection.hs

```diff
@@ -136,71 +147,105 @@ coinMapToList = fmap (uncurry CoinMapEntry) . Map.toList . unCoinMap
 coinMapValue :: CoinMap a -> Coin
 coinMapValue = mconcat . fmap entryValue . coinMapToList
 
--- | Selects an entry at random from a 'CoinMap', returning both the selected
---   entry and the map with the entry removed.
---
--- If the given map is empty, this function returns 'Nothing'.
---
-coinMapRandomEntry
-    :: MonadRandom m
-    => CoinMap a
-    -> m (Maybe (CoinMapEntry a), CoinMap a)
-coinMapRandomEntry (CoinMap m)
-    | Map.null m =
-        return (Nothing, CoinMap m)
-    | otherwise = do
-        ix <- fromEnum <$> generateBetween 0 (toEnum (Map.size m - 1))
-        let entry = uncurry CoinMapEntry $ Map.elemAt ix m
-        let remainder = CoinMap $ Map.deleteAt ix m
-        return (Just entry, remainder)
-
 --------------------------------------------------------------------------------
 -- Coin Selection
 --------------------------------------------------------------------------------
 
--- | Provides a common interface for coin selection algorithms.
+-- | Provides a __common interface__ for coin selection algorithms.
+--
+-- The function 'selectCoins', when applied to the given
+-- 'CoinSelectionParameters' object (with /available inputs/ and /requested/
+-- /outputs/), will generate a 'CoinSelectionResult' (with /remaining inputs/
+-- and a /coin selection/).
+--
+newtype CoinSelectionAlgorithm i o m = CoinSelectionAlgorithm
```

Comment:
Can't this be a type alias instead?

Reply:
> Can't this be a type alias instead?

It could be. Though would there be an advantage to making it an alias?

I'll leave that for a future PR though. I've added an item here: https://github.com/input-output-hk/cardano-coin-selection/issues/30